ansible/roles/sysprep/tasks/main.yml

```diff
@@ -42,6 +42,19 @@
       state: touch
       mode: "{{ machine_id_mode }}"
 
+- name: Truncate D-Bus machine-id
```

Comment:
Please add a comment explaining why we need to do this.

Reply:
Sure. It's in the commit message but I'll drop something in the file itself.

Reply:
Is this only needed for Ubuntu on VMWare?

Reply:
It appears to be the case. My VMware RHEL builds work fine without it, I suspect it's purely an Ubuntu thing. I've only found references to it being a problem on Ubuntu (see the linked vmware KB article in the commit message for an example).

Oddly, I have one older physical Ubuntu 20.04 server at home where /var/lib/dbus/machine-id is already a symlink, perhaps it changed in a point release sometime?

Reply:
I have updated the commit message with another link to https://www.freedesktop.org/software/systemd/man/systemd-machine-id-setup.html which explains how /etc/machine-id can be recreated from the DBUS machine-id if it is not also cleaned. Also added a comment above the relevant blocks in Ansible. 

It does not affect RHEL as on my RHEL base images, /var/lib/dbus/machine-id is also already a symlink. But this is not the case of a stock Ubuntu 20.04 install from the latest ubuntu-20.04.4-live-server-amd64.iso.

Reply:
I think its fair to run this only on the vmware case. can we add a check for this as @dlipovetsky suggests?

Reply:
i think moving to vmware provider makes sense, with a note.

Reply:
Change moved to vmware provider as discussed. Thanks for the feedback!